pinot-spi/src/main/java/org/apache/pinot/spi/data/function/evaluators/ExpressionEvaluator.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.data.function.evaluators;
+
+import java.util.List;
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+/**
+ * Interface for evaluators of transform function expressions of schema field specs
+ * They transformFunction follows the convention:
+ *  "transformFunction": "FunctionType({function}, argument1, argument2,...argumentN)"
+ *  For example,
+ *  "transformFunction" : "Groovy({firstName + ' ' + lastName}, firstName, lastName)"
+ */
+public interface ExpressionEvaluator {
```

Comment:
Is this really `expression` evaluator, or more of a `function` evaluator? For example, the function could be a complex groovy script?  Also, how about nested functions?

Reply:
Yes it can be several expressions, functions, nested functions. 
Function was already quite overloaded. I didn't want this to be confused as an evaluator for TransformFunction. And we also have a FunctionExpressionEvaluator (which should go away).
Besides, this is only invoked inside the ExpressionTransformer, hence thought ExpressionEvaluator makes sense.
I don't have a strong preference. But just thought this was less confusing